src/extension.ts

```diff
@@ -184,6 +192,50 @@ export async function activate(context: vscode.ExtensionContext): Promise<IPower
     };
 }
 
+/** Registers a command that waits for PSES Activation and returns the PID, used to auto-attach the PSES debugger */
+function registerWaitForPsesActivationCommand(context: vscode.ExtensionContext) {
+    return vscode.commands.registerCommand(
+        "PowerShell.WaitForPsesActivationAndReturnProcessId",
+        async () => {
+            const pidFileName = `PSES-${vscode.env.sessionId}.pid`;
+            const pidFile = vscode.Uri.joinPath(context.globalStorageUri, "sessions", pidFileName);
+            const fs = vscode.workspace.fs;
+            // Wait for the file to be created
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition, no-constant-condition
+            while (true) {
+                try {
+                    const pidContent = await fs.readFile(pidFile);
+                    const pid = parseInt(pidContent.toString(), 10);
+                    try {
+                        // Check if the process is still alive, delete the PID file if not and continue waiting.
+                        process.kill(pid, 0);
+                    } catch {
+                        await fs.delete(pidFile);
+                        continue;
+                    }
+                    return pidContent.toString();
```

Comment:
Why return a stringified int out of this function?

Reply:
Because it is referenced in the launch command via the ${command:} syntax and that requires it to be explicity a string return, it will error otherwise (I got bit by this). I will add an appropriate comment.

Reply:
Got it, makes since. Have similarly been bit and forgot.